plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1078,7 +1078,8 @@ private void checkExternalPath(HdfsContext context, Path path)
     {
         try {
             if (!isS3FileSystem(context, hdfsEnvironment, path)) {
-                if (!hdfsEnvironment.getFileSystem(context, path).isDirectory(path)) {
+                FileSystem fileSystem = hdfsEnvironment.getFileSystem(context, path);
+                if (fileSystem.exists(path) && !fileSystem.getFileStatus(path).isDirectory()) {
```

Comment:
nit: change the error message to: `"External location exists, but not a directory: " + path`

Reply:
@phd3 Could you review this again please?